File: examples/runnable/04-schedules/go/report.go
// Package main holds the schedules lab: create a daily Schedule that fires a
// trivial DailyReportWorkflow at 09:00. The Go equivalent of CreateSchedule.java.
package main

import (
	"go.temporal.io/sdk/workflow"
)

const TaskQueue = "reports"

// DailyReportWorkflow is the Workflow a Schedule fires. Kept trivial on purpose:
// the lab is about the Schedule, not the report.
func DailyReportWorkflow(ctx workflow.Context, reportName string) error {
	workflow.GetLogger(ctx).Info("building report", "name", reportName)
	return nil
}

File: examples/runnable/04-schedules/python/report.py
"""The Workflow a Schedule fires. The Python equivalent of DailyReportWorkflow.java.

Kept trivial on purpose: the lab is about the Schedule, not the report. Run a
Worker on TASK_QUEUE if you want the scheduled runs to actually execute.
"""

from temporalio import workflow

TASK_QUEUE = "reports"


@workflow.defn
class DailyReportWorkflow:
    @workflow.run
    async def run(self, report_name: str) -> None:
        workflow.logger.info("building report %s", report_name)
